### PR TITLE
[netcore] Fix Math.ILogB corner cases

### PR DESCRIPTION
**Pull request by EgorBo**
This PR fixes 5 tests in `System.Runtime.Extensions.Tests` (MathTests)
It's basically a copy-paste from CoreCLR:
1) https://github.com/dotnet/coreclr/blob/bd7f8e7435b8cba0cca3a198f15fdc2e4764c2fc/src/pal/src/configure.cmake#L869-L892
2) https://github.com/dotnet/coreclr/blob/master/src/pal/src/cruntime/math.cpp#L876-L894

```
System.Runtime.Extensions.Tests  Total: 6466, Errors: 0, Failed: 11, Skipped: 2
```

### CODE REVIEW

File: netcore/excludes-System.Runtime.Extensions.Tests.rsp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+-nomethod System.Tests.AppDomainTests.FirstChanceException_Called
+
+# AssemblyLoadContext.InternalLoad is not implemented
+-nomethod System.Tests.AppDomainTests.LoadBytes
+
+# TODO: investigate
+-nomethod System.Tests.AppDomainTests.TypeResolve
+-nomethod System.Tests.AppDomainTests.GetAssemblies
+-nomethod System.Tests.AppDomainTests.ResourceResolve
+-nomethod System.Tests.AppDomainTests.AssemblyLoad
```
Comment (filipnavara) [inline netcore/excludes-System.Runtime.Extensions.Tests.rsp]: The events are not wired up from the native side. In .NET Core the `AppDomain` class forwards them to `AssemblyLoadContext`. Mono on the native side still expects `AppDomain` (renamed to `MonoDomain` for .NET Core) which implements none of the events: https://github.com/mono/mono/blob/7082845114f603b3e174c26d9e8fb7f667bc2a51/mcs/class/System.Private.CoreLib/Mono/MonoDomain.cs#L17-L37

File: mono/metadata/sysmath.c [hunk 2]
```diff
@@ -204,7 +204,20 @@ ves_icall_System_Math_Ceiling (gdouble v)
 gint32
 ves_icall_System_Math_ILogB (gdouble x)
 {
-	return ilogb (x);
+	int ret;
+#if !HAVE_COMPATIBLE_ILOGB0
```
Comment (filipnavara) [inline mono/metadata/sysmath.c]: Kinda makes me wonder why FP_ILOGB0 is not tested directly here without all the `autoconf` magic. Apparently it's a constant.

File: mono/metadata/sysmath.c [hunk 3]
```diff
@@ -204,7 +204,20 @@ ves_icall_System_Math_Ceiling (gdouble v)
...
+	if (x == 0.0)
+		ret = -2147483648;
+	else 
+#endif
+
+#if !HAVE_COMPATIBLE_ILOGBNAN
```
Comment (filipnavara) [inline mono/metadata/sysmath.c]: Same here for FP_ILOGBNAN.

File: mono/metadata/sysmath.c (hunk 2 above)
Comment (EgorBo) [reply mono/metadata/sysmath.c]: hm.. makes sense indeed.

File: netcore/excludes-System.Runtime.Extensions.Tests.rsp (hunk 1 above)
Comment (EgorBo) [reply netcore/excludes-System.Runtime.Extensions.Tests.rsp]: Ah ok, so it's basically the same issues as with FirstChanceException event

Comment (filipnavara) [reply netcore/excludes-System.Runtime.Extensions.Tests.rsp]: Yep. I was thinking about fixing it, but I didn't want to step on anyone's toes. I believe @lambdageek was doing some work on `AssemblyLoadContext`, perhaps not directly related to this?

Comment (lambdageek) [reply netcore/excludes-System.Runtime.Extensions.Tests.rsp]: Yea, it's slow going. I got bogged down in refactoring `MonoImage:references` (a list of mono assemblies) to work correctly with ALCs.  I'll try to handoff the work to someone else. but right now it's stuck in a refactoring swamp

Comment (filipnavara) [reply netcore/excludes-System.Runtime.Extensions.Tests.rsp]: This part seems to be quite isolated from the rest of `AssemblyLoadContext` in terms of dependencies. Aside from placing it in the right place in the native code it should not clash much with the rest of the refactoring.

**Comment by marek-safar** [comment]
@lewurm @vargaz could you please review
